### PR TITLE
add from polygon method to OSMRoadNetwork

### PR DESCRIPTION
**Pull request by nreinicke**
In order to make building a new road network easier on the user, this adds a new `from_polygon` class method to the `OSMRoadNetwork` that uses osmnx and shapely on the backend to build a road network.

### CODE REVIEW

**Review by robfitzgerald — CHANGES_REQUESTED** [review]
i love the idea of making it easier, even built-in, to build a road network. that said, i don't love the idea of osmnx being a _required_ dependency. if you can set this up where osmnx (and shapely) are optional dependencies (at the build level) and we do a check at the script level for the existence of the osmnx package (and throw if missing), then i think that would be the right move.

osmnx is like a framework-sized dependency, and getting it installed alongside other dependencies is difficult. i feel the same way about geopandas. i think it would be great to not require either of these.

File: nrel/hive/model/roadnetwork/osm/osm_builders.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    try:
+        import osmnx as ox
+    except ImportError as e:
+        raise ImportError(
+            "the osmnx package is required for building an OSMRoadNetwork from a polygon"
+        ) from e
+
+    ox.settings.all_oneway = True
+
+    G = ox.graph_from_polygon(polygon, network_type="drive")
```
[the remark this answers is not in the document]
Comment (nreinicke) [reply nrel/hive/model/roadnetwork/osm/osm_builders.py:17]: yeah, I set this up such that osmnx only gets imported if this method gets called explicitly, see [here](https://github.com/NREL/hive/blob/c481881faf35b40429b287d654f06fab464a20d4/nrel/hive/model/roadnetwork/osm/osm_builders.py#L9). That way it's only an optional dependency if you want to use the `OSMRoadNetwork.from_polygon()` method (which isn't called at all in any of the core library code, just in something like [this](https://github.com/NREL/hive/blob/ndr/osm_network_build_method/examples/download_road_network.py).

File: nrel/hive/model/roadnetwork/roadnetwork.py [hunk 2]
```diff
@@ -21,7 +21,6 @@ class RoadNetwork(ABC):
     """
 
     sim_h3_resolution: H3Resolution
-    geofence: Optional[GeoFence]
```
[the remark this answers is not in the document]
Comment (nreinicke) [reply nrel/hive/model/roadnetwork/roadnetwork.py:24]: agreed, I can put in an issue for getting rid of this entirely.

File: nrel/hive/model/roadnetwork/osm/osm_builders.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (robfitzgerald) [reply nrel/hive/model/roadnetwork/osm/osm_builders.py:17]: oh, cool, this is exactly what i was asking for. special thanks to 1:52pm Nick for being ahead of his time.

**Review by robfitzgerald — APPROVED** [review]
:butter:

**Pull request merged**